src/twisted/internet/_sslverify.py

```diff
@@ -21,6 +22,109 @@
 from twisted.python import log
 from twisted.python._oldstyle import _oldStyle
 
+# If we don't have TLS v1.3 yet, we can't disable it -- this is just so when it
+# makes it into OpenSSL, connections end up going to v1.3
+if not hasattr(SSL, "OP_NO_TLSv1_3"):
+    _OP_NO_TLSv1_3 = 0x00
+else:
+    _OP_NO_TLSv1_3 = SSL.OP_NO_TLSv1_3
+
+
+
+class TLSVersion(Names):
+    """
+    TLS versions that we can negotiate with the client/server.
+    """
+    SSLv3 = NamedConstant()
+    TLSv1_0 = NamedConstant()
+    TLSv1_1 = NamedConstant()
+    TLSv1_2 = NamedConstant()
+    TLSv1_3 = NamedConstant()
+    Highest = NamedConstant()
+
+
+
+_tlsDisableFlags = {
+    TLSVersion.SSLv3: SSL.OP_NO_SSLv3,
+    TLSVersion.TLSv1_0: SSL.OP_NO_TLSv1,
+    TLSVersion.TLSv1_1: SSL.OP_NO_TLSv1_1,
+    TLSVersion.TLSv1_2: SSL.OP_NO_TLSv1_2,
+    TLSVersion.TLSv1_3: _OP_NO_TLSv1_3,
+}
+
+
+class TLSVersionRange(object):
+    """
+    A range of L{TLSVersion}s.
+    """
+
+    def __init__(self, fromVersion, toVersion):
+        """
+        @param fromVersion: The minimal version to support.
+        @type fromVersion: L{TLSVersion} constant
+
+        @param toVersion: The maximal version to support (where Highest is the
+            newest at all times)
+        @type toVersion: L{TLSVersion} constant
+        """
+        versions = list(TLSVersion.iterconstants())
+
+        if versions.index(fromVersion) > versions.index(toVersion):
```

Comment:
NamedConstants compare for order.  You can just do `fromVersion > toVersion` to express this idea.